test/functional/cursor_tests.js

```diff
@@ -4501,4 +4502,34 @@ describe('Cursor', function() {
 
     testTransformStream(config, done);
   });
+
+  it('should apply parent read preference to count command', function(done) {
```

Comment:
Does this test pass with a replicaset? It's not passing for me locally.

Reply:
How can we imitate your local environment? We're not sure why it would be failing.

Reply:
Here's the error: 
```
1) Cursor should apply parent read preference to count command:
     Uncaught TypeError: self.s.connectOptions.readPreference.equals is not a function
      at shouldTriggerConnect (/Users/katherinewalker/drivers/node/mongodb-core/lib/topologies/replset.js:770:42)
      at /Users/katherinewalker/drivers/node/mongodb-core/lib/topologies/replset.js:841:15
      at applyAuthenticationContexts (/Users/katherinewalker/drivers/node/mongodb-core/lib/topologies/replset.js:731:12)
      at Server.<anonymous> (/Users/katherinewalker/drivers/node/mongodb-core/lib/topologies/replset.js:802:7)
      at /Users/katherinewalker/drivers/node/mongodb-core/lib/topologies/server.js:506:16
      at /Users/katherinewalker/drivers/node/mongodb-core/lib/connection/pool.js:531:18
      at _combinedTickCallback (internal/process/next_tick.js:131:7)
      at process._tickCallback (internal/process/next_tick.js:180:9)
```